pkg/kfapp/kustomize/kustomize.go

```diff
@@ -1422,14 +1422,19 @@ func GenerateYamlWithOperatorAnnotation(resMap resmap.ResMap, instance *unstruct
 					Message: fmt.Sprintf("failed to create corev1 client: %v", err),
 				}
 			}
-			_, err = corev1client.Namespaces().Get(m.GetName(), metav1.GetOptions{})
+			ns, err := corev1client.Namespaces().Get(m.GetName(), metav1.GetOptions{})
 			if err == nil {
 				log.Infof("Namespace %v already exists.", m.GetName())
 
-				_, found := anns[kfdefAnn]
-				if !found || anns[kfdefAnn] != kfdefCr {
-					// if the namespace is not created by this operator, should not append the annotation
+				nsAnns := ns.GetAnnotations()
```

Comment:
Just so I understand, the `anns := m.GetAnnotations()` on line 1409 did not get the namespace annotations properly, so you needed this?

Reply:
correct. That line got the annotation from the kustomize file not the current instance.